swagger-templates/src/main/java/com/okta/swagger/codegen/AbstractOktaJavaClientCodegen.java

```diff
@@ -90,159 +84,141 @@
-                            RefProperty refProperty;
-                            // detect array properties
-                            if (rawSchema instanceof ArrayProperty) {
-                                Property innerProp = ((ArrayProperty) rawSchema).getItems();
-                                if (innerProp instanceof RefProperty) {
-                                    refProperty = (RefProperty) innerProp;
-                                } else {
-                                    // invalid swagger config file
-                                    throw new SwaggerException("Expected 'schema.items.$ref' to exist.");
-                                }
-                            } else if (rawSchema instanceof RefProperty) {
-                                // non array, standard ref property typically in the format of '#/Definitions/MyModel'
-                                refProperty = (RefProperty) rawSchema;
-                            } else {
-                                throw new SwaggerException("Expected 'schema' to be of type 'ArrayProperty' or 'RefProperty'.");
+                        .filter(entry -> "200".equals(entry.getKey()) && entry.getValue().getContent() != null)
+                        .filter(entry -> entry.getValue().getContent().get("application/json") != null)
+                        .forEach(entry -> {
+                            Schema schema = entry.getValue().getContent().get("application/json").getSchema();
+                            if (schema != null && schema instanceof ArraySchema) {
+                                String ref = ((ArraySchema) schema).getItems().get$ref();
+                                resources.add(refToSimpleName(ref));
                             }
-
-                            // get the simple name 'MyModel' instead of '#/Definitions/MyModel'
-                            resources.add(refProperty.getSimpleRef());
-                        }
-                    });
-                })
-        );
+                        });
+                });
+        });
 
         // find any children of these resources
-        swagger.getDefinitions().forEach((name, model) -> {
-            String parent = (String) model.getVendorExtensions().get("x-okta-parent");
-            if (parent != null) {
-                parent = parent.replaceAll(".*/", "");
-
-                if (resources.contains(parent)) {
-                    resources.add(parent);
+        openAPI.getComponents().getSchemas().forEach((name, model) -> {
+            if(model.getExtensions() != null && model.getExtensions().containsKey("x-okta-parent")) {
+                String parent = (String) model.getExtensions().get("x-okta-parent");
+                if (parent != null) {
+                    parent = parent.replaceAll(".*/", "");
+
+                    if (resources.contains(parent)) {
+                        resources.add(parent);
+                    }
```

Comment:
Should really be porting the two changes here: https://github.com/okta/okta-sdk-java/pull/681/files#r779240785